### PR TITLE
ARROW-11449: [CI][R][Windows] Use ccache

### PR DESCRIPTION
**Pull request by kou**
38m -> 20m

### CODE REVIEW

**Comment by kou** [comment]
* Rtools 3.5: 24m: https://github.com/apache/arrow/runs/1805519013
* Rtools 4.0: 20m: https://github.com/apache/arrow/runs/1805519041

**Review by pitrou — APPROVED** [review]
Nice improvement!

File: .github/workflows/r.yml [hunk 1]
```diff
@@ -168,6 +168,22 @@ jobs:
...
+      - name: Setup ccache
+        shell: bash
+        run: |
+          ci/scripts/ccache_setup.sh
+          echo "CCACHE_DIR=$(cygpath --absolute --windows ccache)" >> $GITHUB_ENV
+      # We must enable actions/cache before r-lib/actions/setup-r to ensure
+      # using system tar instead of tar provided by Rtools.
+      # We can use tar provided by Rtools when we drop support for Rtools 3.5.
+      # Because Rtools 4.0 or later has zstd. actions/cache requires zstd
+      # when tar is GNU tar.
```
Comment (nealrichardson) [inline .github/workflows/r.yml:180]: 🤯

**Review by nealrichardson — APPROVED** [review]
I don't know much about ccache but seems like this is working as expected

**Pull request closed without merging**